draft-irtf-cfrg-hpke.md

```diff
@@ -847,12 +847,11 @@ rejection sampling over field elements:
 
 ~~~
 def DeriveKeyPair(ikm):
-  prk = LabeledExtract(zero(0), I2OSP(kem_id, 2), ikm)
+  prk = LabeledExtract(zero(0), concat(I2OSP(kem_id, 2), "dkp:prk"), ikm)
   sk = 0
   counter = 1
   while sk == 0 or sk >= order:
```

Comment:
Can this loop more than 255 times? If so, the encoding of counter might wrap below.

Reply:
Very good question! Maybe @rozbb knows? I'll try to check later, too.

Reply:
I'm pretty sure the probability of this looping more than 255 times is negligible. Nevertheless, maybe we can have a short-circuit in that case? That is, inside the loop:

~~~
if counter > 255:
   ABORT()
~~~

Reply:
Indeed, the likelihood of this looping even twice is negligible. I suppose if we're doing the range check on sk (which is within range with all-but-negligible probability), then it makes sense to do a counter check.

I'm not a huge fan of the exception-y language though. What about `panic("DeriveKeyPair failed")`? I'm bikeshedding at this point, so feel free to use whatever.

Reply:
I used `raise` since it matches the notation used elsewhere. I'd treat this sort of failure the same as a `Context.Seal` failure.

Reply:
Added the check with the most recent commit. Are you fine with having the check directly at the beginning?

Do you like adding the probability of failure? How was it calculated by the way?

Oh, and why does the counter start with 1? If it started with 0, we would win one trial before failing. Very improbable that this one more trial makes a difference, sure ;)